src/extension/utils.ts

```diff
@@ -171,3 +171,50 @@ export async function initWasm(wasmUri: Uri) {
     }
   )
 }
+
+// eslint-disable-next-line max-len
+// adapted from https://github.com/microsoft/vscode/blob/5e4c5c78122d02469ba19d778588ee05fe841f5a/extensions/markdown-language-features/server/src/util/dispose.ts#L41
+export abstract class DisposableRegistrar implements vscode.Disposable {
```

Comment:
It does make sense to me wanting to avoid DRY for classes implementing `vscode.Disposable` to clean up after themselves. However, I think it'd rather move this to an issue to consider brought adoption of the pattern across the codebase. Not saying this is a bad idea, rather consider a consolidated effort to standardize on it than introducing something in a single one place in the code base as part of another effort.

Reply:
Gotcha! I'll move the code from `DisposableRegistrar` into `ShowTerminalProvider` for now, and go ahead and make an issue about wider adoption of the `Disposable` pattern.